src/main/java/org/jvnet/hudson/test/RealJenkinsRuleInit.java

```diff
@@ -27,16 +27,27 @@
 import hudson.Plugin;
 import java.net.URL;
 import java.net.URLClassLoader;
+
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
 import jenkins.model.Jenkins;
 
+/**
+ * Plugin for use by {@link RealJenkinsRule}.
+ * <p>
+ * <stong>NOTE</strong>: this and only this class is added into a dynamically generated plugin, see {@link PluginUtils#createRealJenkinsRulePlugin(java.io.File, String)}.
+ * In order for this to occur correctly there need to be no inner classes or other code dependencies here (except what can be loaded by reflection).
+ */
+@Restricted(NoExternalUse.class)
```

Comment:
This will have no effect since the access control checker only runs on `*-plugin/src/main/java/**/*.java` not `*-plugin/src/test/java/**/*Test.java`. Might be useful as a reminder to humans alongside the Javadoc but that is it. You could consider prefixing the class name with `_` or something to emphasize that code is not intended to link against it.

Reply:
move package to "do_not_consume" ?

Reply:
Maybe, or some `impl` package. I am not particularly concerned. Might suffice to clarify in the Javadoc summary sentence (prior to first `.` IIRC) that test code should not refer to the class.

Reply:
defc0f73e753897698cb9fe3dac086b4461f7fa8